packages/next/src/client/components/react-dev-overlay/internal/helpers/enqueue-client-error.ts

```diff
@@ -0,0 +1,22 @@
+import { isHydrationError } from '../../../is-hydration-error'
+
+// Dedupe the two consecutive errors: If the previous one is same as current one, ignore the current one.
+export function enqueueConsecutiveDedupedError(
+  queue: Array<Error>,
+  error: Error
+) {
+  const isFront = isHydrationError(error)
+  const previousError = isFront ? queue[0] : queue[queue.length - 1]
```

Comment:
previous erorr is for deduping, we have some workaround to still show the hydration errors first like commented below. so if the last error is from front then pick the 1st one in the queue otherwise pick the tail one as the previous error. Then do the comparison.

Otherwise the console.error will take over the place displaying the warning but not correctly treated as hydration error. Like what we discussed [here](https://vercel.slack.com/archives/C04TR3U872A/p1729253606257269). It's sth need to be improved later

Reply:
Can you give me an example of an error sequence and what would happen without this logic. Unclear to me why hydration errors are treated differently.

Reply:
For instance testing against nested p tag: `<p><p>xxx</p></p>`

We'll receive 2 errors

1st is a warning from `console.error`, it look like this, due to lack of better handling of hydration warnings, it's just a normal error with large message, but not determined as a good hydration yet.
![image](https://github.com/user-attachments/assets/a9b466e6-2ce1-479a-ba72-189e55a62f6b)

2nd is the error thrown from react which a hydration error.
![image](https://github.com/user-attachments/assets/bff01c96-8a83-40e6-8533-a25ab16dfcaa)


So currently the workaround is put it in the 1st, then I want to find a generic solution later to also capture that warning as a hydration error as well. It would take some long time to make it

Reply:
I think the issue here is that we don't amend the diff of the first message and instead duplicate the cause in the final hydration error message. The final hydration error message should be left untouched. 

We should only format the diff of in the first message. But this would require a change in how we display hydration messages so we need to leave it for now.